src/apps/frontend/pages/authentication/reset-password/reset-password-form.tsx

```diff
@@ -0,0 +1,130 @@
+        Setup your new password here
+      </h3>
+
+      <form onSubmit={formik.handleSubmit}>
+        <div className="mb-4.5">
+          <FormControl
+            error={formik.touched.password && formik.errors.password}
+            label={'Password'}
+          >
+            <Input
+              data-testid="password"
+              disabled={isResetPasswordLoading}
+              error={formik.touched.password && formik.errors.password}
+              name="password"
+              onBlur={formik.handleBlur}
+              onChange={formik.handleChange}
+              type={isPasswordVisible ? 'text' : 'password'}
+              value={formik.values.password}
+              placeholder="Enter your new password"
+            />
+            <button
+              className="absolute inset-y-0 right-0 flex items-center px-4"
+              onClick={(e) => {
+                e.preventDefault();
+                togglePasswordVisibility();
+              }}
+            >
+              {isPasswordVisible ? (
+                <img
+                  className="size-6.5 opacity-65"
+                  src="/assets/img/icon/eye-closed.svg"
+                  alt="hide password icon"
+                />
+              ) : (
+                <img
+                  className="size-6.5 opacity-65"
+                  src="/assets/img/icon/eye-open.svg"
+                  alt="show password icon"
+                />
+              )}
+            </button>
+          </FormControl>
+        </div>
+        <div className="mb-6">
+          <FormControl
+            error={formik.touched.confirmPassword && formik.errors.confirmPassword}
+            label={'Re-type Password'}
+          >
+            <Input
+              data-testid="confirmPassword"
+              disabled={isResetPasswordLoading}
+              error={formik.touched.confirmPassword && formik.errors.confirmPassword}
+              name="confirmPassword"
+              onBlur={formik.handleBlur}
+              onChange={formik.handleChange}
+              type={isRetypePasswordVisible ? 'text' : 'password'}
+              value={formik.values.confirmPassword}
+              placeholder="Re-enter the password"
+            />
+            <button
+              className="absolute inset-y-0 right-0 flex items-center px-4"
+              onClick={(e) => {
+                e.preventDefault();
+                toggleRetypePasswordVisibility();
+              }}
+            >
+              {isRetypePasswordVisible ? (
+                <img
+                  className="size-6.5 opacity-65"
+                  src="/assets/img/icon/eye-closed.svg"
+                  alt="hide password icon"
+                />
+              ) : (
+                <img
+                  className="size-6.5 opacity-65"
+                  src="/assets/img/icon/eye-open.svg"
+                  alt="show password icon"
+                />
+              )}
+            </button>
```

Comment:
sounds like we need to have password input that toggle password visibility. I see this being repeated